lib/publiccloud/instance.pm

```diff
@@ -91,4 +95,96 @@ sub upload_log {
     assert_script_run("test -d '$tmpdir' && rm -rf '$tmpdir'");
 }
 
+sub wait_for_guestregister
+{
+    my ($self, %args) = @_;
+    $args{timeout} //= 300;
+    my $start_time = time();
+    my $last_info  = 0;
+
+    while (time() - $start_time < $args{timeout}) {
+        my $out = $self->run_ssh_command(cmd => 'sudo systemctl is-active guestregister', proceed_on_failure => 1, quiet => 1);
+        if ($out eq 'inactive') {
+            return time() - $start_time;
+        }
+        if (time() - $last_info > 10) {
+            record_info('WAIT', 'Wait for guest register: ' . $out);
+            $last_info = time();
+        }
+        sleep 1;
+    }
+    die('guestregister didn\'t end in expected timeout=' . $args{timeout});
+}
+
+sub check_ssh_port
+{
+    my ($self, %args) = @_;
+    $args{timeout} //= 600;
+    my $start_time = time();
+
+    while ((my $duration = time() - $start_time) < $args{timeout}) {
+        return $duration if (script_run('nc -vz -w 1 ' . $self->{public_ip} . ' 22', quiet => 1) == 0);
+        sleep 1;
+    }
+    return;
+}
+
+sub get_uptime
```

Comment:
inside ipa.pm we have `extract_startup_timings` . looks like at least partially they doing the same thing . maybe we need to combine/reuse code of this two ?

Reply:
I found that you moved function from ipa but it doesn't change much my question

Reply:
decided to drop a function

Reply:
This function isn't used anymore. So we can remove this method.